### ISSUE TITLE
e-mail notification not sent / global name 'SMTPException' is not defined

### ISSUE THREAD

**Issue by blu-IT** [OnyxfireInc/OnyxBackupVM#20]
Hello,

I tested the script and it works without any problem on XCP-ng server v. 8.1, but at the end of vm backup e-mail notifications are not sent:

```
global name 'SMTPException' is not defined
Traceback (most recent call last):
  File "./onyxbackup-vm.py", line 73, in run
    xenService.send_email()
  File "/OnyxBackupVM/onyxbackup/service/service.py", line 383, in send_email
    except SMTPException as e:
NameError: global name 'SMTPException' is not defined
```

Someone who can give me a hint?

Thanks!

Regards

**Comment by lancefogle** [comment]
"SMTPException as e" needs to read "smtplib.SMTPException as e" in service.py

I will release a fixed version soon after testing.

**Comment by lancefogle** [comment]
@blu-IT Please test latest master commit and let me know if resolved on your end. It looks good from my initial testing.

**Comment by blu-IT** [comment]
@lancefogle 

Hi, thanks!#

Now I get this error:

`(!) Email report failed to send: Connection unexpectedly closed: timed out `

The log says;

`2020-07-20 20:40:50 - ERROR: (!) Email report failed to send: Connection unexpectedly closed: timed out [ service:send_email():384 ]`

Maybe it is a config issue now?

**Comment by lancefogle** [comment]
Great! Thank you for testing.

Yes, that is a client-side error. It is sometimes the case that email servers enforce delays for various reasons (anti-spam, etc) like sendmail does with its "greet pause" feature or it could be a network condition. Either way, that is caused by a socket.timeout exception being raised from smtplib due to the connection process to the SMTP server taking over the default socket.timeout period. I am not setting the timeout in the code currently so it should use the default timeout of your system.

You can see what your default timeout is by executing the following in a python shell:
```
import socket
socket.getdefaulttimeout()
```

**Comment by lancefogle** [comment]
Fixed by 47e085e
